lib/include/public/IECSClient.hpp

```diff
@@ -51,6 +52,10 @@ namespace Microsoft {
 
                     // [optional] enabled ECS telemetry
                     bool enableECSClientTelemetry = false;
+
+                    // [optional] Mandatory agents list. If not present in response, the payload
+                    // is determined as bad
+                    std::unordered_set<std::string> mandatoryAgents;
```

Comment:
Did you miss adding any file, as `mandatoryAgents` is not used anywhere?

Reply:
It will be used in the EXPCommonClient.cpp file in this PR - [Pull 230 - Contributing back patches to 1DS modules](https://github.com/microsoft/cpp_client_telemetry_modules/pull/230/files). I think this PR will have to be merged first before we merged the private one.